zds/tutorialv2/api/views.py

```diff
@@ -154,3 +154,39 @@ def get_queryset(self):
         prefetch_related_objects(exports, "published_object")
 
         return exports
+
+
+class ClapView(CreateAPIView, RetrieveDestroyAPIView):
+    serializer_class = ClapSerializer
+
+    def create(self, request, *args, **kwargs):
+        data = self.add_extra_data(request)
+        serializer = self.serializer_class(data=data)
+
+        if serializer.is_valid():
+            serializer.save()
+            headers = self.get_success_headers(serializer.data)
+            return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def delete(self, request, *args, **kwargs):
+        clap = Clap.objects.filter(publication_id=self.kwargs["publication_id"], user_id=self.request.user.id).first()
+        if clap is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+        clap.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def get(self, request, *args, **kwargs):
+        clap = Clap.objects.filter(publication_id=self.kwargs["publication_id"]).filter(user=self.request.user).first()
```

Comment:
```suggestion
        clap = Clap.objects.filter(publication_id=self.kwargs["publication_id"], user=self.request.user).first()
```
Tu l'as pourtant bien fait juste au-dessus :)